lib/Dialect/Linalg/Transforms/LinalgTransforms.cpp

```diff
@@ -36,8 +36,10 @@ const StringLiteral mlir::linalg::LinalgTransforms::kLinalgTransformMarker =
 LogicalResult mlir::linalg::tileLinalgOpAndSetMarker(PatternRewriter &rewriter,
                                                      Operation *op,
                                                      ArrayRef<int64_t> sizes,
-                                                     StringRef linalgMarker) {
-  auto tileRes = tileLinalgOperation(rewriter, op, sizes);
+                                                     StringRef linalgMarker,
+                                                     ArrayRef<int64_t> permutation) {
+  assert(permutation.size() == 0 || permutation.size() == sizes.size());
```

Comment:
Please document these preconditions in the documentation.

Reply:
Added documentation in LinalgTransforms.h including these preconditions